core/state/gen_allocs.go

```diff
@@ -0,0 +1,25 @@
+package state
+
+import (
+	"math/big"
+
+	"github.com/dominant-strategies/go-quai/core/genallocs"
+	"github.com/dominant-strategies/go-quai/params"
+)
+
+// Will go through the balance schedule and add each one to the account's balance in state.
+func (state *StateDB) AddLockedBalances(blockNum *big.Int, genesisAccounts []genallocs.GenesisAccount) error {
+	uintBlockNum := blockNum.Uint64() - 1 // Off by 1 to include in correct block.
+	// Check if this block is a monthly unlock.
+	if uintBlockNum%params.BlocksPerMonth == 0 {
+		// Rotate through the accounts and apply the unlocks valid for this month.
+		for _, account := range genesisAccounts {
+			accountAddr, err := account.Address.InternalAddress()
+			if err != nil {
+				return err
+			}
+			state.AddBalance(accountAddr, account.BalanceSchedule[uintBlockNum])
```

Comment:
this index also feels is off by one because we store the balance schedule on the key that is a multiple of blocks per month -1, lets say blockNum is 1201, blocks per month is 100, we will enter this if statement, and store the balance schedule under 1200. 			

but mathematically this index is always one less than the multiple of the blocks per month, which means index will be 1999, 1399, 1599, etc. So the uintBlockNum used in the if statement can never match the index? 
```
account.BalanceSchedule[(12+i)*params.BlocksPerMonth-1] = quaiPerUnlock // Off by 1 to include in the following block.
```